packages/commons/src/events/FieldConfig.ts

```diff
@@ -594,16 +594,28 @@ const HttpField = BaseField.extend({
     method: z.enum(['GET', 'POST', 'PUT', 'DELETE']),
     headers: z.record(z.string()).optional(),
     body: z.record(z.string()).optional(),
-    params: z.record(z.string()).optional(),
+    params: z.record(z.string(), FieldReference).optional(),
     timeout: z
       .number()
       .default(15000)
       .describe('Request timeout in milliseconds')
   })
-}).describe('HTTP request function triggered by a button click')
+}).describe('HTTP request function triggered by a button click or other event')
 
 export type HttpField = z.infer<typeof HttpField>
 
+const QueryParamReaderField = BaseField.extend({
+  type: z.literal(FieldType.QUERY_PARAM_READER),
+  configuration: z.object({
+    formProjection: z
+      .record(z.string())
+      .optional()
+      .describe('Projection of the field value after parsing the query string')
+  })
+})
```

Comment:
```suggestion
}).describe('A field that maps URL query params into form values and clears them afterward')
```
Or something that fits more. Should help out with the documentation